libs/langgraph/langgraph/store/store.py

```diff
@@ -0,0 +1,99 @@
+from typing import Any, Optional
+from weakref import WeakKeyDictionary
+
+from langgraph.store.base import BaseStore, GetOp, Item, PutOp, SearchOp
+
+CACHE: WeakKeyDictionary[BaseStore, "Store"] = WeakKeyDictionary()
+
+
+def _get_store(store: BaseStore) -> "Store":
+    if store not in CACHE:
+        CACHE[store] = Store(store)
+    return CACHE[store]
+
+
+class Store:
+    __slots__ = ("_store",)
+
+    def __init__(self, store: BaseStore) -> None:
+        self._store = store
+
+    def get(
+        self,
+        namespace: tuple[str, ...],
+        id: str,
+    ) -> Optional[Item]:
+        return self._store.get([GetOp(namespace, id)])[0]
+
+    def search(
+        self,
+        namespace_prefix: tuple[str, ...],
+        /,
+        *,
+        query: Optional[str],
+        filter: Optional[dict[str, Any]],
+        weights: Optional[dict[str, float]],
+        limit: int = 10,
+        offset: int = 0,
+    ) -> list[Item]:
+        return self._store.search(
+            [
+                SearchOp(namespace_prefix, query, filter, weights, limit, offset),
+            ]
+        )[0]
+
+    def put(
+        self,
+        namespace: tuple[str, ...],
+        id: str,
```

Comment:
Ambiguity for me: does the ID have to be globally unique? Or unique just within a namespace? I'm assuming global rn

Reply:
I think the name is fine; maybe just a docstring clarification?
don't want it unnecessarily verbose

Reply:
Added docs